### DISCUSSION TITLE
Inconsistent behavior when extending a application generator versus generating an app using the CLI

### DISCUSSION THREAD

**Discussion by beeman** [nrwl/nx#23302] [General]
### Current Behavior

In [this project](https://github.com/solana-developers/create-solana-dapp) we extend the Next and React generators to create apps.

In a default Nx 18 and 19 project (with Crystal) the app that gets generated by our extended generators, the content of `project.json` is verbose as can be seen in [this generated app](https://github.com/beeman/solana-dapp-react-version/blob/main/web/project.json).

However, if we generate a second app using the `nx` CLI in the same workspace, the content of `project.json` is (mostly) using the inferred targets as can be seen [here](https://github.com/beeman/solana-dapp-react-version/blob/main/second/project.json).

The curious thing is that this is happening without having set `useInferencePlugins` in `nx.json` or having the env var set.

### Expected Behavior

Considering the workspace does not have `useInferencePlugins` set, I would expect the extended generator to behave the same as when we're using the CLI.

### GitHub Repo

https://github.com/beeman/solana-dapp-react-version

### Steps to Reproduce

1. Create a new app using `pnpx create-solana-dapp@next`
2. Pick the React option (for consistency with the example, same happens with Next though).
3. Verify that `web/project.json` is verbose
4. Create a second app: `pnpm nx g @nx/react:app second --e2e-test-runner none --routing false`
5. Verify that `second/project.json` is inferred.

### Nx Report

```shell
❯ pnpm nx report

 NX   Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.

Pass --verbose to see the stacktraces.

❯ pnpm nx reset

 NX   Resetting the Nx workspace cache and stopping the Nx Daemon.

This might take a few minutes.


 NX   Daemon Server - Stopped


 NX   Successfully reset the Nx workspace.

❯ pnpm nx report

 NX   Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.

Pass --verbose to see the stacktraces.
```

### Package Manager Version

pnpm 9.0.5

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

This also happened with Nx18 but I decided to wait till Nx19 to make sure it wasn't fixed there.

Somehow, `nx report` fails in my demo project, I'll create a separate issue for that (https://github.com/nrwl/nx/issues/23296).

**Comment by Coly010** [comment]
Hey @beeman!

Perhaps there is a misunderstanding here on Inferred Projects with new workspaces.
By default, inferred targets is turned on.

The `useInferencePlugins` and env var are options provided to allow people to specifically opt-out of inferred targets. Therefore when our first party plugins (such as `@nx/react`) are run, they will attempt to use inferred targets, even if other projects in the workspace are using defined projects in project.json files.

I'm going to convert this issue to a discussion because this is correct behaviour but perhaps it should be discussed more.

**Comment by beeman** [comment]
Hi @Coly010, thanks for the quick reply!

I indeed think I don't understand how it's supposed to work. 

As you can see [here](https://github.com/solana-developers/create-solana-dapp/blob/main/packages/preset-next/src/utils/generate-next-application.ts#L8-L18), we are using first-party plugins to generate these things. We wrap them, but the actual generator is the first party plugin. 

I would expect this to work just like how it's when I invoke `nx g @nx/react:app`. 

The important part is that I expect this to work predictably and consistently. Currently, it's none. 

I don't want to force people into disabling inferred plugins as I think it makes sense for people to get a generated project that's as similar as if they would have run the commands themself (create-nx-workspace, install @nx/react, then generate the app). That's why I don't set the `useInferencePlugins` flag, nor set the env var. Still the 2 options behave differently.

Somehow, this is not the case when I extend, but it is the case when I manually invoke it which causes my confusion.
